components/script/dom/node.rs

```diff
@@ -230,6 +230,10 @@ bitflags! {
         /// Whether this node has a weird parser insertion mode. i.e whether setting innerHTML
         /// needs extra work or not
         const HAS_WEIRD_PARSER_INSERTION_MODE = 1 << 11;
+
+        /// Whether this node has a serve as the text container for editable content of
+        /// <input> or <textarea> element.
+        const IS_TEXT_EDITING_ROOT = 1 << 13;
```

Comment:
“text editing root” might be a confusing name. I suggest renaming it to "text control inner editor".